parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/FeatureAnalysers.scala

```diff
@@ -0,0 +1,224 @@
+
+  val outcode: FeatureName = "outcode"
+  /**
+    * @return true if the string is in the Tokens.outcodes collection, false if not
+    */
+  def outcodeAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.outcodes)
+
+  val postTown: FeatureName = "posttown"
+  /**
+    * @return true if the string is in the Tokens.postTown collection, false if not
+    */
+  def postTownAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.postTown)
+
+  val flat: FeatureName = "flat"
+  /**
+    * @return true if the string is in the Tokens.flat collection, false if not
+    */
+  def flatAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.flat)
+
+  val company: FeatureName = "company"
+  /**
+    * @return true if the string is in the Tokens.company collection, false if not
+    */
+  def companyAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.company)
+
+  val road: FeatureName = "road"
+  /**
+    * @return true if the string is in the Tokens.road collection, false if not
+    */
+  def roadAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.road)
+
+  val residential: FeatureName = "residential"
+  /**
+    * @return true if the string is in the Tokens.residential collection, false if not
+    */
+  def residentialAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.residential)
+
+  val business: FeatureName = "business"
+  /**
+    * @return true if the string is in the Tokens.business collection, false if not
+    */
+  def businessAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.business)
+
+  val locational: FeatureName = "locational"
+  /**
+    * @return true if the string is in the Tokens.locational collection, false if not
+    */
+  def locationalAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.locational)
+
+  val ordinal: FeatureName = "ordinal"
+  /**
+    * @return true if the string is in the Tokens.ordinal collection, false if not
+    */
+  def ordinalAnalyser(): CrfFeatureAnalyser[Boolean] = ContainsAnalyser(Tokens.ordinal)
+
+  val hyphenations: FeatureName = "hyphenations"
+  /**
+    * @return the count of characters which are '-'
+    */
+  def hyphenationsAnalyser(): CrfFeatureAnalyser[Int] = CrfFeatureAnalyser[Int](_ count(_ == '-'))
+
+  val digits: FeatureName = "digits"
+  /**
+    * @return a DigitLiteral String, which indicates if the string has all digits, contains digits or no digits
+    */
+  def digitsAnalyser(): CrfFeatureAnalyser[String] = {
+    import DigitsLiteral._
+    CrfFeatureAnalyser[String] { str =>
+      str.allDigits[String] { rs =>
+        if(rs) {
+          allDigits
+        } else {
+          str.containsDigits[String] { rs =>
+            if(rs) {
+              someDigits
+            } else {
+              noDigits
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * ref digitsAnalyser
+    */
+  object DigitsLiteral {
+    val allDigits = "all_digits"
```

Comment:
For future reference it's neither ALL CAPS, nor camelCase: http://stackoverflow.com/a/9752606/555355